src/kixi/stats/distribution.cljc

```diff
@@ -654,30 +661,38 @@
   "Returns a beta distribution.
   Params: {:n ∈ ℕ > 0, :alpha ∈ ℝ > 0, :beta ∈ ℝ > 0}"
   [{:keys [n alpha beta] :or {alpha 1.0 beta 1.0}}]
+  (assert (pos-int? n) (str "n (" n ") must be a positive integer."))
+  (assert (and (pos? alpha) (pos? beta)) (str "alpha (" alpha ") and beta (" beta ") must be positive."))
   (->BetaBinomial n alpha beta))
 
 (defn weibull
   "Returns a weibull distribution.
   Params: {:shape ∈ ℝ >= 0, :scale ∈ ℝ >= 0}"
   [{:keys [shape scale] :or {shape 1.0 scale 1.0}}]
+  (assert (and (not (neg? shape)) (not (neg? scale)))
+          (str "shape (" shape ") and scale (" scale ") must not be negative."))
   (->Weibull shape scale))
 
 (defn chi-squared
   "Returns a chi-squared distribution.
   Params: {:k ∈ ℕ > 0}"
   [{:keys [k]}]
+  (assert (pos-int? k) (str "k (" k ") must be a positive integer."))
   (->ChiSquared k))
 
 (defn f
   "Returns an F distribution.
   Params: {:d1 ∈ ℕ > 0, :d2 ∈ ℕ > 0}"
   [{:keys [d1 d2]}]
+  (assert (and (pos-int? d1) (pos-int? d2))
```

Comment:
Parameters are often integers, but they don't have to be.

Reply:
Okay.  I'll update the docstring as well.